modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenPasswordGrantTestCase.java

```diff
@@ -250,7 +288,21 @@ public void testGetAccessTokenWithPasswordGrant() throws Exception {
         assertNotNull(jwtClaims);
     }
 
-    @Test(groups = "wso2.is", description = "Verify the presence of the updated scopes in the access token")
+    @Test(groups = "wso2.is", dependsOnMethods = "testGetAccessTokenWithPasswordGrant", description =
+            "Verify the pre issue access token action request")
+    public void testPreIssueAccessTokenActionRequest() throws Exception {
+
+        String actualRequestPayload = actionsMockServer.getReceivedRequestPayload(MOCK_SERVER_ENDPOINT_RESOURCE_PATH);
+        PreIssueAccessTokenActionRequest actualRequest =
+                new ObjectMapper().readValue(actualRequestPayload, PreIssueAccessTokenActionRequest.class);
+
+        PreIssueAccessTokenActionRequest expectedRequest = getRequest();
```

Comment:
Shouldn't we validate this in other grant types as well ? If so let's move the `getRequest()` to the ActionBaseTestCase class

Reply:
Request can differ based on grant types. So let's keep this here. We might need to find what's common and fix accordingly